### PR TITLE
Add  native toolbar

### PR DESCRIPTION
**Pull request by skress**
This change adds a native toolbar ... and a couple of other things as well. Not sure if you like the added dependencies :-)

- Updated dependencies
- Added @webview-bridge/react-native to have a typed interface between the native and the web-view part
- Added react-native-keyboard-controller which provides a way to customize the toolbar atop the virtual keyboard (works cross-platform) - requires reanimated as peer dependency
- react-native-keyboard-controller requires native extensions which were not support on Expo Go, so had to switch that off as well
- Changed initialization of the editor state, instead of sending a message, I am now setting a global JS variable that's used during editor initialization
- Only kept the size change messages, moved rest to typed interface via webview-bridge

I have added some placeholder text above the editor to see what happens when the keyboard pops up. Thanks to react-native-keyboard-controller, the screen scrolls up on iOS. It does not work on Android. There is a text input field above the editor - for that field the screen scrolls up on Android as well. So the problem seems to be related to the web-view integration.

### CODE REVIEW

**Comment by strdr4605** [comment]
Hey @skress, thanks for your PR. Had a quick look and it seems nice. As there are a lot of changes, it may take some time to do a review. Will try to do a deeper dive in the following week.

**Review by strdr4605 — APPROVED** [review]
@skress 

Looked into the code. Overall it looks ok. Maybe a bit too many changes and engineering😅.

Wondering if we should add a changelog in README, to point to the initial version (from the article). And new additions, like your changes?

Also, can you add a demo gif to README?

**Comment by skress** [comment]
So, yeah. I am also not really happy with such a large PR. On the other hand I guess it would have been a bit difficult to split it up as that would have required us to discuss the intermediate steps. At this early point of the project that seemed problematic as well :-)

I will add a changelog and a demo gif - will try to do that tomorrow.

**Comment by skress** [comment]
Sorry, took a bit longer. I've also updated the version in package.json to 1.1.0 (to have something to reference in the changelog). If there is anything else you'd like me to change, just let me know.

File: README.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
 # react-native-lexical
 
-This playground project uses Lexical editor in React Native through react-native-weview.
+This playground project uses Lexical editor in React Native through react-native-webiew.
```
Comment (strdr4605) [inline README.md:3]: typo:  react-native-webview

File: README.md [hunk 2]
```diff
@@ -11,8 +11,23 @@ npm i
...
 
 Also if you have any ideas, questions, or problems, you can open an issue in this repo and we can discuss it.
+
+## Changelog
+
+- [1.0.0](https://github.com/Planable/react-native-lexical/tree/912158dc092040af3ca7935ab2d555cac5ff1f6a)
+  - Initial version which corresponds to what was explained in the [blog post](https://strdr4605.com/how-to-set-up-lexical-editor-in-react-native)
+- 1.1.0 (current version)
+  - Added native toolbar (above keyboard)
+  - Added a couple of helper libraries (e.g. webview-bridge, react-native-keyboard-controller) to ease development and make the showcase a bit more realistic
```
Comment (strdr4605) [inline README.md:33]: Please reverse the order.

- 1.1.0
  - ...
- 1.0.0
  - ...

**Comment by strdr4605** [comment]
Hey @skress, thanks again for the contribution🙌. Hope that people will profit from it and make their Mobile apps better.

**Pull request merged**